### PR TITLE
Fix API

### PR DESCRIPTION
**Pull request by northern-64bit**
# Description

Implement a new and improved API

# How has this been tested?

* Please describe the tests that you ran to verify your changes. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.

# Checklist:

- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [ ] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: openbb_terminal/api.py [hunk 1]
```diff
@@ -62,59 +98,131 @@ def __call__(self, *args, **kwargs):
...
+            spec = None
+
+        if spec is None:
+            return None
+        else:
+            module = importlib.import_module(module_path)
+
+            return module
 
     @staticmethod
```
Comment (Chavithra) [inline openbb_terminal/api.py:172]: If you need the object attributes/methods it's a normal method.
If you need to use the class properties/functions inside the function it's better to turn it into a `@classmethod`. 
If you don't need anything related to the object or the class you can turn it into a `@staticmethod`.

Here I would turn it into a `@classmethod`.

**Review by Chavithra — CHANGES_REQUESTED** [review]
Maybe we can start using that on reports to see how it behaves.

We might also need to find more explicit names for the following classes:
- `APIFactory` 
- `Item`

File: openbb_terminal/api.py (hunk 1 above)
Comment (northern-64bit) [reply openbb_terminal/api.py]: Done!

**Comment by minhhoang1023** [comment]
Will we be able to get code introspection as discussed in the call?

Even though less desirable, docstring would still contain the information users need. 
 
The disadvantage I see is that we would need to put in more effort to maintain docstrings. Previously, we could get away with having an extra or missing param in the docstring. Without the code introspective, docstring is the only information users have without having to go through the source code. So we really need to make sure docstring is correct and up-to-date.

**Comment by northern-64bit** [comment]
> Is there anything we can do @Chavithra @northern-64bit to expose the args and kwargs that the children have through the factory?

This is now fixed
image

**Pull request closed without merging**